dataclients/kubernetes/definitions/definitions.go

```diff
@@ -43,3 +44,16 @@ type WeightedBackend interface {
 	GetName() string
 	GetWeight() float64
 }
+
+func errorsJoin(errs ...error) error {
```

Comment:
So its only used from the methods that are never called by the Skipper.
We should add a test for it.

Reply:
Actually they aren't called anywhere

Reply:
Makes sense to note that we are adding this because we can not use https://pkg.go.dev/errors#Join before 1.21 release.
If we stick to "\n" delimiter then we can add a TODO to use stdlib errors.Join

Reply:
PR message (or better commit message) should note that original `err = errors.Wrap(err, nerr.Error())` is wrong because it returns nil when err is nil.